### DISCUSSION TITLE
onValidate event or similar

### DISCUSSION THREAD

**Discussion by hvri5h** [react-hook-form/react-hook-form#1941] [Q&A] [answered]
I'm using a yup schema for my form:
```
  const { control, handleSubmit, errors, watch } = useForm({
    mode: 'onBlur',
    defaultValues,
    validationSchema,
  });
```

And rendering inputs with `Controller` and MUI components like so:
```
<Controller
    as={TextField}
    control={control}
    error={!!errors.confirmEmail}
    label="Confirm email"
    name="confirmEmail"
    helperText={errors.confirmEmail ? errors.confirmEmail?.message : null}
/>
```

I'm trying to find a way to show an input adornment (green tick) to a textfield on successful validation.

`onChange` or `onBlur` events both get triggered before validation and using `rules` means I have to write my own validation logic which I don't want to do.

Is there a way trigger a piece of code on some kind of validation event without writing a custom resolver?

**Comment by bluebill1049** [comment, marked as answer]
`triggerValidation`?

**Reply by hvri5h** [reply]
Sorry, my question wasn't clear enough. 

I'm aware of `triggerValidation`, but is there a callback for this function? I want to execute code when validation is completed

**Reply by bluebill1049** [reply]
yes you can `await` for it and call your action

**Reply by hvri5h** [reply]
Ah, of course! Thank you :)

**Comment by kansoft** [comment]
But what about if we want to check if all fields (form) are validated? I need to do call action after user click submit button and there is errors in the form. Any ideas?

**Reply by Moshyfawn** [reply]
I may be misunderstanding your question, but from what I gather, you are describing the default behavior of RHF. If you call `handleSubmit` in the form's `onSubmit` callback or the button's `onClick`, the submission will validate the form and display any error messages.

**Reply by kansoft** [reply]
@Moshyfawn yes, of course, the behaviour of RHF is clear for me. The form is validating when user click submit button, but how to react on this validation? I want to know when validation is done. I need that to do additional action besides form submission.
I found that `formState` object is available in `useForm` and I can use `isValidating` field to check that. But it would be great to have something like o`onValidate` in RHF.

**Comment by karan042** [comment]
Is there any workaround for this?
